hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -1450,9 +1537,8 @@ private void sendReplicateCommand(final ContainerInfo container,
     final ContainerID id = container.containerID();
     final ReplicateContainerCommand replicateCommand =
         new ReplicateContainerCommand(id.getId(), sources);
-    inflightReplication.computeIfAbsent(id, k -> new ArrayList<>());
     sendAndTrackDatanodeCommand(datanode, replicateCommand,
-        action -> inflightReplication.get(id).add(action));
+        action -> inflightReplication.add(id, action));
```

Comment:
I guess we should probably check the return status here, and increment some metric if we are skipping the replication for now?

Reply:
Sure, let's add some metric.